one/src/ethereum_rpc.rs

```diff
@@ -0,0 +1,175 @@
+                "jsonrpc": "2.0",
+                "method": "eth_getBlockByHash",
+                "params": [block_hash, false],
+                "id": 1,
+            }))
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+
+    async fn eth_transaction_by_hash(&self, transaction_hash: &str) -> Result<serde_json::Value> {
+        // Get the block_hash and input from the transaction.
+        // curl https://mainnet.infura.io/v3/{api_token} \
+        //   -X POST \
+        //   -H "Content-Type: application/json" \
+        //   -d '{"jsonrpc":"2.0","method":"eth_getTransactionByHash",
+        //        "params":["0xBF7BC715A09DEA3177866AC4FC294AC9800EE2B49E09C55F56078579BFBBF158"],"id":1}'
+        // >> {"jsonrpc":"2.0", "id":1, "result": {
+        //       "blockHash": "0x783cd5a6febe13d08ac0d59fa7e666483d5e476542b29688a6f0bec3d15febd4",
+        //       "blockNumber": "0x105f34f",
+        //       "input": "0x97ad09eb41b6408c1b4be5016f652396ef47c0982c36d5877ebb874919bae3a9b854d8e1"
+        //    }}
+        Ok(self
+            .client
+            .post(&self.url)
+            .json(&serde_json::json!({
+                "jsonrpc": "2.0",
+                "method": "eth_getTransactionByHash",
+                "params": [transaction_hash],
+                "id": 1,
+            }))
+            .send()
+            .await?
+            .json()
+            .await?)
+    }
+}
+
+impl EthRpc for HttpEthRpc {
+    async fn root_time_by_transaction_cid(&self, cid: Cid) -> Result<RootTime> {
+        // transaction to blockHash, blockNumber, input
+        let tx_hash = format!("0x{}", hex::encode(cid.hash().digest()));
+        let json: serde_json::Value = self.eth_transaction_by_hash(&tx_hash).await?;
+        debug!("txByHash response: {}", json);
+        let result = json.get("result").context("missing field result")?;
+        let block_hash = result
+            .get("blockHash")
+            .context("missing field result.blockHash")?;
+        let block_hash = block_hash
+            .as_str()
+            .context("missing field result.blockHash was not a string")?;
+        let block_number = result
+            .get("blockNumber")
+            .context("missing field result.blockNumber")?;
+        let block_number = block_number
+            .as_str()
+            .context("missing field result.blockNumber was not a string")?;
+        let block_number = get_timestamp_from_hex_string(block_number)
+            .context("missing field result.blockNumber was not a hex timestamp")?;
+        let input = result.get("input").context("missing field result.input")?;
+        let input = input
+            .as_str()
+            .context("missing field result.input was not a string")?;
+        let root = get_root_from_input(input)?;
+
+        // Get the block with block_hash if latest block number minus result.number grater then 3 the block is stable.
```

Comment:
this is interesting.  This is a new check, I'm pretty sure js-ceramic doesn't do this when validating TimeEvents.

I'm not necessarily opposed, just surprised we're changing the logic here.  This is also going to create an extra call to the eth rpc endpoint and increase the latency of validating time events.  Is this something we want to do right now or maybe leave as a possible future improvement?